### PR TITLE
pkp/pkp-lib#2816 Allow publication date to be set via UI

### PR DESCRIPTION
**Pull request by NateWr**

### CODE REVIEW

File: controllers/tab/catalogEntry/form/CatalogEntryCatalogMetadataForm.inc.php [hunk 1]
```diff
@@ -121,6 +122,7 @@ function initData() {
 			'licenseURL' => $submission->getDefaultLicenseURL(),
 			'arePermissionsAttached' => !empty($copyrightHolder) || !empty($copyrightYear) || !empty($licenseURL),
 			'confirm' => ($this->_publishedMonograph && $this->_publishedMonograph->getDatePublished())?true:false,
+			'datePublished' => $submission->getDatePublished(),
```
Comment (bozana) [inline controllers/tab/catalogEntry/form/CatalogEntryCatalogMetadataForm.inc.php]: I believe it is not necessary to provide datePublished here, when it is provided in the fetch function -- the fetch function is/should be always called at the end, after the init function?

File: classes/services/SubmissionService.inc.php [hunk 2]
```diff
@@ -100,7 +100,8 @@ public function addToCatalog($submission) {
 		$submission->setStatus(STATUS_PUBLISHED);
 		\Application::getSubmissionDao()->updateObject($submission);
 
-		$publishedMonograph->setDatePublished(\Core::getCurrentDate());
+		$datePublished = $submission->getDatePublished() ? $submission->getDatePublished() : \Core::getCurrentDate();
+		$publishedMonograph->setDatePublished($datePublished);
```
Comment (bozana) [inline classes/services/SubmissionService.inc.php]: Isn't submission->get/setDatePublished() = $publishedMonograph->get/setDatePublished() ? If so, then maybe just to set the date to the current date if there is not date?

Comment (NateWr) [reply classes/services/SubmissionService.inc.php]: Maybe I'm confusing myself, but I think they're different objects here. `$publishedMonograph` is either pulled from the database based on the submission ID or a new `$publishedMonograph` is created. In either case, the `$submission` could have a new value, and if so this should be used to update the `$publishedMonograph`. Is that right?

File: controllers/tab/catalogEntry/form/CatalogEntryCatalogMetadataForm.inc.php (hunk 1 above)
Comment (NateWr) [reply controllers/tab/catalogEntry/form/CatalogEntryCatalogMetadataForm.inc.php]: :+1:

File: classes/services/SubmissionService.inc.php (hunk 2 above)
Comment (bozana) [reply classes/services/SubmissionService.inc.php]: Ah ok, sorry, I thought wrong :-)

Comment (bozana) [reply classes/services/SubmissionService.inc.php]: please merge if ready :-)

**Pull request merged**